src-java/kilda-persistence-tinkerpop/src/test/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepositoryTest.java

```diff
@@ -91,8 +94,49 @@ public void setUp() {
 
         switchA = createTestSwitch(TEST_SWITCH_A_ID.getId());
         switchB = createTestSwitch(TEST_SWITCH_B_ID.getId());
+        switchC = createTestSwitch(TEST_SWITCH_C_ID.getId());
 
-        assertEquals(2, switchRepository.findAll().size());
+        assertEquals(3, switchRepository.findAll().size());
+    }
+
+    @Test
+    public void findPortToFlowsMap() {
+        createTestFlow(TEST_FLOW_ID, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_2, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_3, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_4, switchC, switchB);
+
+        Map<Integer, List<Flow>> result = flowRepository.findSwitchFlowsByPort(switchA.getSwitchId(), null);
+
+        assertNotNull(result);
+        assertTrue("The map must contain a key for the port", result.containsKey(PORT_1));
+        assertNotNull("The map must contain a non-null value for the test port", result.get(PORT_1));
+        assertEquals("The map must contain exactly two keys", 2, result.size());
+        assertEquals("The map must contain exactly 3 flows", 3, result.get(PORT_1).size());
+        assertTrue("The map must contain exactly 3 flows",
+                result.get(PORT_1).stream().map(Flow::getFlowId).collect(Collectors.toList())
+                .containsAll(Arrays.asList(TEST_FLOW_ID, TEST_FLOW_ID_2, TEST_FLOW_ID_3)));
+    }
+
+    @Test
+    public void findPortToFlowsMapWithFilter() {
```

Comment:
same. I'd change the name to something like "shouldFindSwitchFlowsByPortWithFilter" to keep the naming convection in this class

Reply:
as we discussed on one of the past syncs, we might want to revisit the naming style. I suggest to leave it for now as-is. Once we arrive to a conclusion, I'll update the names